src/main/java/org/openstreetmap/atlas/geography/atlas/sub/SubAtlasCreator.java

```diff
@@ -401,6 +401,166 @@ public SubAtlasCreator(final Atlas atlas)
         return Optional.ofNullable(result);
     }
 
+    @SuppressWarnings("squid:S3776")
+    @Override
+    public Optional<Atlas> silkCut(final GeometricSurface boundary)
```

Comment:
I know this has been discussed but I do wonder if it is worth diving into the `SubAtlasCreator` class and factoring out common behaviour. These methods are huge and a lot of the stuff seems similar.